### DISCUSSION TITLE
Issue with https on prod server

### DISCUSSION THREAD

**Discussion by alucard29** [filamentphp/filament#4747] [Help]
Hi ! 
Since I migrated my app (but its just a blank project for now) when I try to access the admin login page, I have issues both with Chrome & Firefox. On Chrome, messages are :
- GET https://myapp/filament/assets/app.css?id=74b0e5d1ac0bd5c2e7700a3ab38b4202 net::ERR_HTTP2_PROTOCOL_ERROR 200
- GET https://myapp/filament/assets/app.js?id=a89643d28638f3978429678585b1ce2d net::ERR_HTTP2_PROTOCOL_ERROR 200

No issue when in http only. Any advice ?
Thx a lot

**Comment by danharrin** [comment]
Are you behind a proxy like Cloudflare?

**Reply by atapatel** [reply]
@danharrin what should i do if application is running behind cloudflare?

**Reply by danharrin** [reply]
Ensure you edit the TrustProxies middleware with `$proxies = ['*']` or the Cloudflare IPs

**Reply by atapatel** [reply]
Already added.

**Reply by atapatel** [reply]
actuall all the asset are loading except filament/asset/app.js.

**Reply by danharrin** [reply]
`Url::forceScheme('https')` in a service provider? APP_URL env?

**Reply by atapatel** [reply]
.env  for APP_URL is 
 `https://dev-api.ecommerce.app`

**Reply by atapatel** [reply]
it is working file locally but there is some issue with production depyment.

**Reply by atapatel** [reply]
2023/01/20 11:55:30 [error] 5#5: *30 upstream timed out (110: Operation timed out) while reading upstream, client: 10.244.1.224, server: _, request: "GET /filament/assets/app.js?id=a4db6005d7b6129ad492a8b48e706423 HTTP/1.1", upstream: "fastcgi://unix:/var/run/php-fpm.sock:",

**Comment by alucard29** [comment]
Hi,
No, in principle

**Reply by danharrin** [reply]
Tried `Url::forceScheme('https')` in a service provider?

**Reply by alucard29** [reply]
not better.

**Reply by alucard29** [reply]
I had same issue with livewire until I publish assets using 
`php artisan livewire:publish --assets`
But I don't know how doing this with Filament

**Reply by danharrin** [reply]
https://github.com/livewire/livewire/issues/242#issuecomment-948054946

**Reply by alucard29** [reply]
Hi !
Thank you ! Even if the proposed solution did not exactly apply to me, it put me on the right track.

**Comment by ArtMin96** [comment]
If you are using nginx: [https://filamentphp.com/tricks/fix-filament-assets-nginx](https://filamentphp.com/tricks/fix-filament-assets-nginx)

**Reply by atapatel** [reply]
@ArtMin96  - I had tried. but no luck :(

**Reply by atapatel** [reply]
Actually php fpm got disconnected while reading app.js via filament controller.

**Comment by Parables** [comment]
Found a solution at https://stackoverflow.com/a/48298029

**Comment by slim8** [comment]
Hi everyone,

If you're facing issues with Livewire or Filament assets not loading properly in production (especially over HTTPS), you can try this solution:

1- Open your routes/web.php file.
2- Add the following code:
_________________________________________________
use Illuminate\Support\Facades\Artisan;

Route::get('/publish-livewire-assets', function () {
    // Run the vendor:publish command for Livewire assets
    Artisan::call('vendor:publish', [
        '--force' => true,
        '--tag' => 'livewire:assets',
    ]);

    return response()->json([
        'message' => 'Livewire assets published successfully.',
    ]);
});
_______________________________________________________

3- Save the file and access the following link in your browser:
https://your-domain.com/publish-livewire-assets
This will publish the Livewire assets and should resolve the issue.

Important:
Make sure to secure or remove this route once the issue is fixed, as it allows sensitive commands to be executed.

I hope this helps! 😊

**Comment by Binyamkpaul** [comment]
i got encounter with proxy sites like this Uncaught ReferenceError: isRecordSelected is not defined
    at [Alpine] {
            'hidden': false && isGroupCollapsed(null),
            'bg-gray-50 dark:bg-white/5': isRecordSelected('213817'),
            '[&>*:first-child]:relative [&>*:first-child]:before:absolute [&>*:first-child]:before:start-0 [&>*:first-child]:before:inset-y-0 [&>*:first-child]:before:w-0.5 [&>*:first-child]:before:bg-primary-600 [&>*:first-child]:dark:before:bg-primary-500': isRecordSelected('213817'),
        } (eval at <anonymous> (https://away.doamin.et/proxy/0022b59f/https/subdomain.domain.et/livewire/livewire.min.js?id=38dc8241:5:668),

**Reply by karan218** [reply]
I have been same issue on filament V2 suddenly alpine code not working on production after deployment it's working on local and other dev env.. but not work on production.. not opening any modal or any filter or anything... please help it's very urgent

**Reply by karan218** [reply]
@Binyamkpaul have got any solution ?

**Reply by Binyamkpaul** [reply]
@karan218  NO i didn't found any solution please if there is let me know

**Reply by Binyamkpaul** [reply]
@danharrin Please Sir help us on this issue
